src/Core/Checkout/Promotion/Util/PromotionCodeService.php

```diff
@@ -90,7 +90,12 @@ public function generateIndividualCodes(string $pattern, int $amount, array $cod
         return array_diff($codes, $codeBlacklist);
     }
 
-    public function addIndividualCodes(string $promotionId, int $amount, Context $context): void
+    /**
+     * @throws PromotionException
+     *
+     * @return string[]
+     */
+    public function addIndividualCodes(string $promotionId, int $amount, Context $context): array
```

Comment:
as you can see on the pipeline this is definitely a BC break, so we need to think about another solution

Reply:
But in what sense is it a break?

As there is no interface on the class, the only way it could break, would be when one extends this class, however this should "actually" not happen? :thinking:

Reply:
sounds like a plan :+1: 
wdyt @shopware/product-cc-checkout ?